src/main/java/cst438_2/service/WeatherService.java

```diff
@@ -0,0 +1,57 @@
+package cst438_2.service;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.TimeZone;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+import com.fasterxml.jackson.databind.JsonNode;
+import cst438_2.domain.TimeAndTemp;
+
+@Service
+public class WeatherService {
+	private static final Logger log =
+			LoggerFactory.getLogger(WeatherService.class);
+	private RestTemplate restTemplate;
+	private String weatherUrl;
+	private String apiKey;
+	
+	public WeatherService(
+			@Value("${weather.url}") final String weatherUrl,
+			@Value("${weather.apikey}") final String apiKey) {
+		this.restTemplate = new RestTemplate();
+		this.weatherUrl = weatherUrl;
+		this.apiKey = apiKey;
+	}
+	
+	public TimeAndTemp getTimeAndTemp(String cityName) {
+		ResponseEntity<JsonNode> response =
+				restTemplate.getForEntity(
+						weatherUrl + "?q=" + cityName + "&appid=" + apiKey, 
+						JsonNode.class);
+		JsonNode json = response.getBody();
+		log.info("Status code from weather server:" +
+				 response.getStatusCodeValue());
+		double temp = Math.round((json.get("main").get("temp").asDouble() - 273.15) * 9.0 / 5.0 + 32.0);
+		long time = json.get("dt").asLong();
+		int timezone = json.get("timezone").asInt();
+		String convertedTime = convertTime(time, timezone);
+		return new TimeAndTemp(temp,convertedTime,timezone);
+	}
+	
+	public String convertTime(long time, int timezone) {
```

Comment:
Nice way to solve this problem.  Consider adding it to your time and temp and allow for that class to do the formating.